x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rest/RestGetInferenceModelAction.java

```diff
@@ -72,10 +71,7 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
     public Set<String> supportedCapabilities() {
         Set<String> capabilities = new HashSet<>();
         capabilities.add(DEFAULT_ELSER_2_CAPABILITY);
-        if (INFERENCE_METADATA_FIELDS_SYSTEM_PROPERTY.isEnabled()) {
-            capabilities.add(INFERENCE_METADATA_FIELDS_CAPABILITY);
-        }
-
+        capabilities.add(INFERENCE_METADATA_FIELDS_CAPABILITY);
         return Collections.unmodifiableSet(capabilities);
```

Comment:
We can update this method to use `Set.of()` now that it returns a static set of capabilities